packages/origin-247-certificate/src/offchain-certificate/synchronize/strategies/batch/events.batch-queue.ts

```diff
@@ -0,0 +1,46 @@
+import { SynchronizableEvent } from '../../../repositories/CertificateEvent/CertificateEvent.repository';
+import { flatten } from 'lodash';
+
+type CertificateId = number;
+
+export class EventsBatchQueue {
+    private eventsToProcess: Record<CertificateId, SynchronizableEvent[]> = {};
+
+    constructor(events: SynchronizableEvent[]) {
+        const sortedEvents = events.sort((a, b) => a.createdAt.getTime() - b.createdAt.getTime());
+
+        sortedEvents.forEach((event) => {
```

Comment:
Just use `groupBy` from lodash :D

Reply:
I wanted to be sure it's preserving the order, but the tests will catch it anyway, so you're right